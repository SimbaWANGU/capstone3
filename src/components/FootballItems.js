/* eslint-disable react/prop-types */
import React from 'react';
import '../styles/FootballItem.scss';
import ArrowCircleRightOutlinedIcon from '@mui/icons-material/ArrowCircleRightOutlined';

function FootballItems({ name, goalsScored, photo }) {
  const goals = `${goalsScored} Goals`;

  const backgroundColor = '#5788e3b3';
  const backgroundColorGradient = '#5788e380';

  const backgroundImageStyle = {
    backgroundImage: `linear-gradient(to left, ${backgroundColor}, ${backgroundColorGradient}), url(${photo})`,
  };

  return (
    <div className="footballItem" style={backgroundImageStyle}>
      <ArrowCircleRightOutlinedIcon style={{ fill: '#FAFAFA' }} className="arrowGo" />
      <h4>{name}</h4>
      <p>{goals}</p>
    </div>
  );
}

export default FootballItems;
