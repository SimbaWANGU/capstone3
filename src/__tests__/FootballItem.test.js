import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import FootballItems from '../components/FootballItems';

describe('Home page body section tests', () => {
  const testData = {
    photo: 'https://photo.example.com',
    name: 'Example',
    id: 1,
    goalsScored: 5,
  };

  it('Component renders as expected', () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <FootballItems
            photo={testData.photo}
            name={testData.name}
            id={testData.id}
            goalsScored={testData.goalsScored}
          />
        </BrowserRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Component is visible to the user', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <FootballItems
            photo={testData.photo}
            name={testData.name}
            id={testData.id}
            goalsScored={testData.goalsScored}
          />
        </BrowserRouter>
      </Provider>,
    );
    expect(getByTestId('footballItemDisplay')).toBeInTheDocument();
  });
});
