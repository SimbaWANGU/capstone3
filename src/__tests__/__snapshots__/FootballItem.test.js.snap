// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page body section tests Component renders as expected 1`] = `
<div
  className="footballItem"
  data-testid="footballItemDisplay"
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #5788e3b3, #5788e380), url(https://photo.example.com)",
    }
  }
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrowGo css-i4bv87-MuiSvgIcon-root"
    data-testid="ArrowCircleRightOutlinedIcon"
    focusable="false"
    style={
      Object {
        "fill": "#FAFAFA",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
    />
  </svg>
  <h4>
    Example
  </h4>
  <p>
    5 Goals
  </p>
</div>
`;
